### ISSUE TITLE
Recursive mutex not destroyed properly test_energi

### ISSUE THREAD

**Issue by RyanLucchese** [energicryptocurrency/energi#147]
When the `test_energi` tests finish running, a `boost::recursive_mutex` produces an error about being unable to be destroyed properly, after which the program segfaults.

```
*** 181 failures detected in test suite "Energi Test Suite"
test_energi: /usr/include/boost/thread/pthread/recursive_mutex.hpp:104: boost::recursive_mutex::~recursive_mutex(): Assertion `!pthread_mutex_destroy(&m)' failed.
Aborted (core dumped)
```

**Comment by RyanLucchese** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Recursive mutex not destroyed properly test_energi](https://trello.com/c/2g5fq4aH/95-recursive-mutex-not-destroyed-properly-testenergi)

**Comment by RyanLucchese** [comment]
Fixed by #179
